diff --git a/libtar/Android.mk b/libtar/Android.mk
index c3bb691..8873024 100755
--- a/libtar/Android.mk
+++ b/libtar/Android.mk
@@ -14,7 +14,7 @@ LOCAL_C_INCLUDES += external/libselinux/include
 LOCAL_SHARED_LIBRARIES += libselinux
 
 ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
-    LOCAL_STATIC_LIBRARIES += libvold libscrypt_static
+    LOCAL_STATIC_LIBRARIES += libvold libscrypt_static libasync_safe
     LOCAL_SHARED_LIBRARIES +=  \
         android.hardware.boot@1.0 \
         android.hardware.confirmationui@1.0 \
@@ -23,10 +23,10 @@ ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
         android.hardware.keymaster@4.0 \
         android.hardware.keymaster@4.1 \
         android.hardware.weaver@1.0 \
-        android.security.apc-ndk_platform \
-        android.system.keystore2-V1-ndk_platform \
-        android.security.authorization-ndk_platform \
-        android.security.maintenance-ndk_platform \
+        android.security.apc-ndk \
+        android.system.keystore2-V4-ndk \
+        android.security.authorization-ndk \
+        android.security.maintenance-ndk \
         libselinux \
         libbinder_ndk \
         libext4_utils \
@@ -53,7 +53,9 @@ ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
         libhwbinder \
         libchrome \
         libbootloader_message \
-        libgatekeeper_aidl
+        libgatekeeper_aidl \
+        libsysutils \
+        libboot_control_client
 
     LOCAL_CFLAGS += -DUSE_FSCRYPT
     ifeq ($(TW_USE_FSCRYPT_POLICY), 1)
diff --git a/libtar/extract.c b/libtar/extract.c
index be989b0..fcd842f 100755
--- a/libtar/extract.c
+++ b/libtar/extract.c
@@ -570,7 +570,7 @@ tar_extract_dir(TAR *t, const char *realname)
 		uint8_t binary_policy[size];
 		memset(&binary_policy, 0, size);
 
-		if (!lookup_ref_tar(t->th_buf.fep, &binary_policy[0])) {
+		if (!lookup_ref_tar(get_policy_descriptor(t->th_buf.fep), &binary_policy[0])) {
 			LOG("error looking up fscrypt policy for '%s' - %s\n", realname, descriptor);
 			policy_lookup_error = true;
 		}
diff --git a/libtar/libtar.h b/libtar/libtar.h
index f42b201..51db894 100755
--- a/libtar/libtar.h
+++ b/libtar/libtar.h
@@ -22,6 +22,7 @@
 
 #ifdef USE_FSCRYPT
 #include "fscrypt_policy.h"
+#include "fscrypt_compat.h"
 #endif
 
 #ifdef __cplusplus
